src/main/java/io/github/programminglife2016/pl1_2016/parser/nodes/SegmentParser.java

```diff
@@ -51,6 +51,11 @@ public SegmentParser(InputStream positions, InputStream metadata) {
         lastIndices = new HashMap<>();
         SpecimenParser specimenParser = new SpecimenParser();
         this.specimens = specimenParser.parse(metadata);
+        Subject ref = new Specimen();
+        ref.setNameId(REFERENCE);
+        if (specimens != null) {
+            specimens.put(REFERENCE, ref);
+        }
```

Comment:
This part is not necessary any more: the missing genomes are inserted while parsing the segments.